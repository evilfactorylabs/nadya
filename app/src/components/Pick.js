import React, { useState } from 'react'
import DayjsUtils from '@date-io/dayjs'
import BackIcon from '@material-ui/icons/ArrowBack'

import {
  Dialog,
  AppBar,
  Toolbar,
  IconButton,
  Typography,
  CardActions,
  CardContent,
  Button,
  Card,
  Slide,
  TextField,
  InputLabel,
  InputAdornment,
  FormControl,
  OutlinedInput,
  MenuItem,
  Select
} from '@material-ui/core'
import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers'
import { makeStyles } from '@material-ui/core/styles'
import { addSubscription } from 'services/subscription'

const Transition = React.forwardRef((props, ref) => {
  return <Slide direction='left' ref={ref} {...props} />
})

const useStyles = makeStyles(theme => ({
  appBar: {
    position: 'relative'
  },
  input: {
    margin: '0.5rem',
    marginTop: '1rem',
    marginBottom: '1rem'
  },
  title: {
    marginLeft: theme.spacing(2),
    flex: 1
  },
  action: {
    marginLeft: 'auto',
    marginRight: theme.spacing(1),
    marginBottom: theme.spacing(2)
  }
}))

export default function Pick({
  handleTopLevelClose,
  handleBack,
  activeSubscription
}) {
  const classes = useStyles()
  const [cost, setCost] = useState('')
  const [costInterval, setCostInterval] = useState(0)
  const [startSubscription, setStartSubscription] = useState(new Date())

  const shouldButtonDisabled = () =>
    !cost || !costInterval || !startSubscription

  const handleAddSubscription = () => {
    addSubscription({
      serviceId: activeSubscription.id,
      title: activeSubscription.title,
      period: costInterval,
      cost: cost,
      firstBill: startSubscription
    })
      .then(_ => handleTopLevelClose())
      .catch(err => {
        window.alert(err)
        throw err
      })
  }

  return (
    <Dialog
      fullScreen
      open
      onClose={handleBack}
      TransitionComponent={Transition}
    >
      <AppBar className={classes.appBar}>
        <Toolbar>
          <IconButton
            edge='start'
            color='inherit'
            onClick={handleBack}
            aria-label='Close'
          >
            <BackIcon />
          </IconButton>
          <Typography variant='h6' className={classes.title}>
            Tambahkan {activeSubscription.title}
          </Typography>
        </Toolbar>
      </AppBar>
      <Card square={true}>
        <CardContent>
          <FormControl fullWidth margin='normal'>
            <TextField
              name='cost'
              type='number'
              label='Biaya'
              margin='dense'
              variant='outlined'
              value={cost}
              onChange={e => setCost(e.target.value)}
              InputProps={{
                startAdornment: (
                  <InputAdornment position='start'>Rp </InputAdornment>
                )
              }}
            />
          </FormControl>
          <FormControl variant='outlined' fullWidth margin='dense'>
            <InputLabel variant='outlined'>Periode penagihan</InputLabel>
            <Select
              value={costInterval}
              onChange={e => setCostInterval(e.target.value)}
              input={<OutlinedInput labelWidth={150} margin='none' />}
            >
              <MenuItem value={1}>Bulanan</MenuItem>
              <MenuItem value={2}>Tahunan</MenuItem>
            </Select>
          </FormControl>
          <MuiPickersUtilsProvider utils={DayjsUtils}>
            <DatePicker
              fullWidth
              margin='normal'
              label='Tanggal pertama berlangganan'
              inputVariant='outlined'
              value={startSubscription}
              onChange={setStartSubscription}
            />
          </MuiPickersUtilsProvider>
        </CardContent>
        <CardActions>
          <Button
            className={classes.action}
            color='primary'
            variant='contained'
            disabled={shouldButtonDisabled()}
            onClick={handleAddSubscription}
          >
            Simpan
          </Button>
        </CardActions>
      </Card>
    </Dialog>
  )
}
