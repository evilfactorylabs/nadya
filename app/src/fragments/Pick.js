import React from 'react'
import Button from '@material-ui/core/Button'

import { Link } from '@reach/router'

export default () => (
  <>
    <h2>Pick Services</h2>
    <Link to='/pick/spotify'>
      <Button color='primary' variant='outlined'>
        Spotify
      </Button>
    </Link>
  </>
)
