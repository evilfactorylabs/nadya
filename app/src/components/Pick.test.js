import Pick from './Pick'

afterAll(async () => {
  await cleanup()
})

describe('Fragments: Pick.js', () => {
  test('It should render Pick screen', () => {
    const handleTopLevelClose = () => {}
    const handleBack = () => {}
    const activeSubscription = { title: 'Spotify' }
    const { container } = render(
      <Pick
        handleBack={handleBack}
        handleTopLevelClose={handleTopLevelClose}
        activeSubscription={activeSubscription}
      />
    )

    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
