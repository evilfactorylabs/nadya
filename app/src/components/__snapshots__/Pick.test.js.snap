// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragments: Pick.js It should render Pick screen 1`] = `
<div
  aria-hidden="true"
/>
`;
